### PR TITLE
Separating classes for better testability

### LINKED ISSUE #38: Come up with test cases for all the APIs
Unit tests for all API endpoints. Currently none of the APIs have test coverage :-(

### PR DESCRIPTION
**Pull request by jmprathab**
Adding Unit tests

## 🚀 Description

- [x] Separating classes for better testability
- [x] Code reformat and copyright update
- [x] Add sample unit test which uses mock JWT parser implementation(`HealthCheckControllerTest.java`)

## 🧪 How Has This Been Tested?

* `gradlew build` succeeded

## 📦 Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality) - Closes #38
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [x] My code follows the code style of this project(Do your best to follow code styles. If none apply just skip this).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/com/myhome/security/WebSecurity.java [hunk 1]
```diff
@@ -26,24 +27,29 @@
...
+  private final UserDetailsService userDetailsService;
   private final PasswordEncoder passwordEncoder;
+  private final UserDetailFetcher userDetailFetcher;
+  private final AppJwtEncoderDecoder appJwtEncoderDecoder;
 
   public WebSecurity(Environment environment,
-      ObjectMapper objectMapper,
-      AppUserDetailsService appUserDetailsService,
-      PasswordEncoder passwordEncoder) {
+      ObjectMapper objectMapper, UserDetailsService userDetailsService,
```
Comment (mslowiak) [inline src/main/java/com/myhome/security/WebSecurity.java]: RequiresArgsConstructor can be used here

File: src/main/java/com/myhome/services/springdatajpa/CommunitySDJpaService.java [hunk 2]
```diff
@@ -91,7 +90,7 @@ public CommunitySDJpaService(
   @Override
   public Set<String> addHousesToCommunity(String communityId, Set<CommunityHouse> houses) {
     if (!getCommunityDetailsById(communityId).isPresent()) return new HashSet<>();
-      houses.forEach(communityHouse -> communityHouse.setHouseId(generateUniqueId()));
+    houses.forEach(communityHouse -> communityHouse.setHouseId(generateUniqueId()));
```
Comment (mslowiak) [inline src/main/java/com/myhome/services/springdatajpa/CommunitySDJpaService.java]: That operation can be done in line 96 to not loop over collection twice

File: src/test/java/com/myhome/controllers/HealthCheckControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+import static org.hamcrest.Matchers.containsString;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@ActiveProfiles(profiles = "test")
+class HealthCheckControllerTest {
```
Comment (mslowiak) [inline src/test/java/com/myhome/controllers/HealthCheckControllerTest.java]: Looks like integration test is here. Can we rename Test to IntegrationTest?

**Review by mslowiak — CHANGES_REQUESTED** [review]

File: src/main/java/com/myhome/security/WebSecurity.java (hunk 1 above)
Comment (jmprathab) [reply src/main/java/com/myhome/security/WebSecurity.java]: Done

File: src/main/java/com/myhome/services/springdatajpa/CommunitySDJpaService.java (hunk 2 above)
Comment (jmprathab) [reply src/main/java/com/myhome/services/springdatajpa/CommunitySDJpaService.java]: Done

File: src/test/java/com/myhome/controllers/HealthCheckControllerTest.java (hunk 3 above)
Comment (jmprathab) [reply src/test/java/com/myhome/controllers/HealthCheckControllerTest.java]: Done

**Review by mslowiak — APPROVED** [review]

**Pull request merged**